x-pack/legacy/plugins/ml/server/routes/system.ts

```diff
@@ -0,0 +1,204 @@
+          // return that security is disabled and don't call the privilegeCheck endpoint
+          return response.ok({
+            body: {
+              securityDisabled: true,
+              upgradeInProgress,
+            },
+          });
+        } else {
+          const body = request.body;
+          const resp = await context.ml!.mlClient.callAsCurrentUser('ml.privilegeCheck', { body });
+          resp.upgradeInProgress = upgradeInProgress;
+          return response.ok({
+            body: resp,
+          });
+        }
+      } catch (error) {
+        return response.customError(wrapError(error));
+      }
+    })
+  );
+
+  router.get(
+    {
+      path: '/api/ml/ml_capabilities',
+      validate: {
+        query: schema.object({
+          ignoreSpace: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const ignoreSpaces = request.query && request.query.ignoreSpaces === 'true';
+        // if spaces is disabled force isMlEnabledInSpace to be true
+        const { isMlEnabledInSpace } =
+          spacesPlugin !== undefined
+            ? spacesUtilsProvider(spacesPlugin, (request as unknown) as Request)
+            : { isMlEnabledInSpace: async () => true };
+
+        const { getPrivileges } = privilegesProvider(
+          context.ml!.mlClient.callAsCurrentUser,
+          xpackMainPlugin,
+          isMlEnabledInSpace,
+          ignoreSpaces
+        );
+        return response.ok({
+          body: await getPrivileges(),
+        });
+      } catch (error) {
+        return response.customError(wrapError(error));
+      }
+    })
+  );
+
+  router.get(
+    {
+      path: '/api/ml/ml_node_count',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        // check for basic license first for consistency with other
+        // security disabled checks
+        if (isSecurityDisabled(xpackMainPlugin)) {
+          return response.ok({
+            body: await getNodeCount(context),
+          });
+        } else {
+          // if security is enabled, check that the user has permission to
+          // view jobs before calling getNodeCount.
+          // getNodeCount calls the _nodes endpoint as the internal user
+          // and so could give the user access to more information than
+          // they are entitled to.
+          const requiredPrivileges = [
+            'cluster:monitor/xpack/ml/job/get',
+            'cluster:monitor/xpack/ml/job/stats/get',
+            'cluster:monitor/xpack/ml/datafeeds/get',
+            'cluster:monitor/xpack/ml/datafeeds/stats/get',
+          ];
+          const body = { cluster: requiredPrivileges };
+          const resp = await context.ml!.mlClient.callAsCurrentUser('ml.privilegeCheck', { body });
+
+          if (requiredPrivileges.every(p => !!resp.cluster[p])) {
```

Comment:
we can now use `resp.has_all_requested` for checks like this.

Reply:
https://github.com/elastic/kibana/pull/57835/commits/e60e183aecb29f087a418637eea683ce7f84e3c7